### PR TITLE
Implement basic.recover method, closes #138

### LINKED ISSUE #138: Support request for basic.recover
It seems that the [basic.recover][1] command is not available in this extension. Could this be implemented?

As a related question: would it be a good idea to add some generic method to send a command to the AMQP server? As a way to be able to use commands that have not yet been implemented. 

[1]: http://www.rabbitmq.com/amqp-0-9-1-quickref.html#basic.recover

### PR DESCRIPTION
**Pull request by pinepain**

### CODE REVIEW

**Comment by pinepain** [comment]
@lstrojny What about including this to 1.6.0? Implementation of `basic.recover` is pretty trivial, atomic and has no side effect on other code base.

**Comment by lstrojny** [comment]
Sounds good to me. Can you add a changelog entry?

**Comment by pinepain** [comment]
Already done (https://github.com/pinepain/php-amqp/commit/7f9bd072d2211edfa53e105cc290df64abc61e89#diff-34ae68f4adad56c25c5bc05dcb64794eR43).

**Comment by pinepain** [comment]
Done.

File: amqp_channel.c [hunk 1]
```diff
@@ -679,6 +679,56 @@ PHP_METHOD(amqp_channel_class, getConnection)
...
 
+/* {{{ proto bool amqp::basicRecover([bool requeue=TRUE])
+Redeliver unacknowledged messages */
+PHP_METHOD(amqp_channel_class, basicRecover)
+{
+	zval *id;
+	amqp_channel_object *channel;
+	amqp_connection_object *connection;
+
+    amqp_rpc_reply_t res;
```
Comment (lstrojny) [inline amqp_channel.c]: indent?

File: amqp_channel.c [hunk 2]
```diff
@@ -679,6 +679,56 @@ PHP_METHOD(amqp_channel_class, getConnection)
...
+		return;
+	}
+
+	/* Get the channel object out of the store */
+	channel = (amqp_channel_object *)zend_object_store_get_object(id TSRMLS_CC);
+
+	connection = AMQP_GET_CONNECTION(channel);
+	AMQP_VERIFY_CONNECTION(connection, "Could not redeliver unacknowledged messages.");
+
+    amqp_basic_recover(
```
Comment (lstrojny) [inline amqp_channel.c]: indent?

Comment (pinepain) [inline amqp_channel.c]: Fixed.

**Comment by lstrojny** [comment]
Awesome!

**Comment by pinepain** [comment]
Thanks!

**Pull request merged**